avm/client/src/avmHelpers.ts

```diff
@@ -0,0 +1,129 @@
+import { CallResultsArray, InterpreterResult, CallRequest } from './types';
+
+const decoder = new TextDecoder();
+const encoder = new TextEncoder();
+
+export function prepareArgs(
+    air: string,
+    prevData: Uint8Array,
+    data: Uint8Array,
+    params: {
+        initPeerId: string;
+        currentPeerId: string;
+    },
+    callResults: CallResultsArray,
+): string {
+    const callResultsToPass: any = {};
+    for (let [k, v] of callResults) {
+        callResultsToPass[k] = {
+            ret_code: v.retCode,
+            result: v.result,
+        };
+    }
+
+    const paramsToPass = {
+        init_peer_id: params.initPeerId,
+        current_peer_id: params.currentPeerId,
+    };
+
+    const encoded = encoder.encode(JSON.stringify(callResultsToPass));
+
+    const avmArg = JSON.stringify([
+        // force new line
+        air,
+        Array.from(prevData),
+        Array.from(data),
+        paramsToPass,
+        Array.from(encoded),
+    ]);
+
+    return avmArg;
+}
+
+export function convertInterpreterResult(rawResult: string): InterpreterResult {
+    let result: any;
+    try {
+        result = JSON.parse(rawResult);
+    } catch (ex) {
+        throw 'call_module result parsing error: ' + ex + ', original text: ' + rawResult;
+    }
+
+    if (result.error !== '') {
+        throw 'call_module returned error: ' + result.error;
+    }
+
+    result = result.result;
+
+    const callRequestsStr = decoder.decode(new Uint8Array(result.call_requests));
+    let parsedCallRequests;
+    try {
+        if (callRequestsStr.length === 0) {
+            parsedCallRequests = {};
+        } else {
+            parsedCallRequests = JSON.parse(callRequestsStr);
+        }
+    } catch (e) {
+        throw "Couldn't parse call requests: " + e + '. Original string is: ' + callRequestsStr;
+    }
+
+    let resultCallRequests: Array<[key: number, callRequest: CallRequest]> = [];
+    for (const k in parsedCallRequests) {
+        const v = parsedCallRequests[k];
+
+        let arguments_;
```

Comment:
Do we need underscore here?

Reply:
`arguments` is a reserved word in JavaScript used to refer to function arguments. And you are not allowed to assign to in in `'strict mode'`

Reply:
I see, thanks